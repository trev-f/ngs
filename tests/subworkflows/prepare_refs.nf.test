nextflow_workflow {

    name "Test Workflow Prepare_Refs"
    script "subworkflows/prepare_refs.nf"
    workflow "Prepare_Refs"

    test("Prepare_Refs successfully unzips and emits GZIP-compressed genome and GTF files.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa.gz"
                input[1] = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf.gz"
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 2

            assert workflow.out.genome
            with(workflow.out.genome) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa/
                assert path(get(0)).md5 == "e050100d0a9a7d06a8accf8424ad0d98"
            }

            assert workflow.out.annotations
            with(workflow.out.annotations) {
                assert size() == 1
                assert get(0) ==~ /^.*\/annotations_I.gtf/
                assert path(get(0)).md5 == "61525d86cb937e0fe5b349215c50498c"
            }

            assert snapshot(workflow.out).match()
        }


    }
    test("Prepare_Refs successfully emits uncompressed genome and GTF files when uncompressed inputs are given.") {

        when {
            params {
                publishDirReports = "${outputDir}/reports"
                publishDirData    = "${outputDir}/data"
            }
            workflow {
                """
                input[0] = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                input[1] = "https://github.com/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                """
            }
        }

        then {
            assert workflow.success
            assert workflow.trace.succeeded().size() == 0

            assert workflow.out.genome
            with(workflow.out.genome) {
                assert size() == 1
                assert get(0) ==~ /^.*\/genome_I.fa/
            }

            assert workflow.out.annotations
            with(workflow.out.annotations) {
                assert size() == 1
                assert get(0) ==~ /^.*\/annotations_I.gtf/
            }

            assert snapshot(workflow.out).match()
        }

    }

}
