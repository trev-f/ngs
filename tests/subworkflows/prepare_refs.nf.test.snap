{
    "Prepare_Refs successfully unzips and emits GZIP-compressed fasta and GTF files.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "1": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "annotations": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "fasta": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-08-01T21:26:23+0000"
    },
    "successfully emits uncompressed fasta and GTF files when uncompressed inputs are given.": {
        "content": [
            {
                "0": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ],
                "1": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "annotations": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "fasta": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ]
            }
        ],
        "timestamp": "2023-08-01T21:26:23+0000"
    },
    "successfully unzips and emits GZIP-compressed fasta and GTF files.": {
        "content": [
            {
                "0": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ],
                "1": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "annotations": [
                    "annotations_I.gtf:md5,61525d86cb937e0fe5b349215c50498c"
                ],
                "fasta": [
                    "genome_I.fa:md5,e050100d0a9a7d06a8accf8424ad0d98"
                ]
            }
        ],
        "timestamp": "2023-08-01T21:26:23+0000"
    },
    "Prepare_Refs successfully emits uncompressed fasta and GTF files when uncompressed inputs are given.": {
        "content": [
            {
                "0": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ],
                "1": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "annotations": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/annotations_I.gtf"
                ],
                "fasta": [
                    "/trev-f/SRAlign-test/raw/rnaseq/data/references/R64-1-1/genome_I.fa"
                ]
            }
        ],
        "timestamp": "2023-08-01T21:26:23+0000"
    }
}